src/placeholder/solvers/interface.py

```diff
@@ -0,0 +1,92 @@
+
+class ModelNotDefinedError(Exception):
+    pass
+
+
+class SolverNotDefinedError(Exception):
+    pass
+
+
+class Solver:
+
+    def __init__(self, model_instance=None):
```

Comment:
Can you add type hints to the class definitions? That way `sphinx` will pick it up automatically :)

Reply:
yes! you mean type hints for like `model_instance`?

Reply:
yes! like `model_instance: Optional[Union[Model, List[ModelList]]] = None` something like that